app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/KeepAccessToSAController.scala

```diff
@@ -0,0 +1,120 @@
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.views.html.templates.ErrorTemplate
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class KeepAccessToSAController @Inject()(
+  authAction: AuthAction,
+  multipleAccountsOrchestrator: MultipleAccountsOrchestrator,
+  mcc: MessagesControllerComponents,
+  val logger: EventLoggerService,
+  sessionCache: TEASessionCache,
+  keepAccessToSA: KeepAccessToSA,
+  val errorView: ErrorTemplate
+)(implicit config: AppConfig, ec: ExecutionContext)
+    extends FrontendController(mcc)
+    with I18nSupport
+    with ControllerHelper {
+
+  implicit val baseLogger: Logger = Logger(this.getClass.getName)
+
+  def view(): Action[AnyContent] =
+    authAction.async { implicit request =>
+      val res = for {
+        _ <- multipleAccountsOrchestrator
+          .checkValidAccountTypeRedirectUrlInCache(
+            List(SA_ASSIGNED_TO_OTHER_USER)
+          )
+        optFormData <- EitherT.right[TaxEnrolmentAssignmentErrors](
+          sessionCache.getEntry[KeepAccessToSAThroughPTA](
+            KEEP_ACCESS_TO_SA_THROUGH_PTA_FORM
+          )
+        )
+      } yield
+        optFormData match {
+          case Some(data) =>
+            KeepAccessToSAThroughPTAForm.keepAccessToSAThroughPTAForm
+              .fill(data)
+          case None => KeepAccessToSAThroughPTAForm.keepAccessToSAThroughPTAForm
+        }
+
+      res.value.map {
+        case Right(form) => Ok(keepAccessToSA(form))
+        case Left(error) =>
+          handleErrors(error, "[KeepAccessToSAController][view]")
+      }
+    }
+
+  def continue: Action[AnyContent] = authAction.async {
+    implicit requestWithUserDetails =>
+      KeepAccessToSAThroughPTAForm.keepAccessToSAThroughPTAForm.bindFromRequest
+        .fold(
+          formWithErrors => {
+            Future.successful(BadRequest(keepAccessToSA(formWithErrors)))
+          },
+          keepAccessToSA => {
+            val res = if (keepAccessToSA.keepAccessToSAThroughPTA) {
+              multipleAccountsOrchestrator
+                .checkValidAccountTypeRedirectUrlInCache(
+                  List(SA_ASSIGNED_TO_OTHER_USER)
+                )
+                .map(_ => routes.SignInWithSAAccountController.view)
+            } else {
+              multipleAccountsOrchestrator
+                .checkValidAccountTypeAndEnrolForPT(SA_ASSIGNED_TO_OTHER_USER)
+                .map(_ => routes.EnrolledPTWithSAOnOtherAccountController.view)
+            }
+
+            res.value.flatMap {
+              case Right(call) =>
+                sessionCache
+                  .save[KeepAccessToSAThroughPTA](
+                    KEEP_ACCESS_TO_SA_THROUGH_PTA_FORM,
+                    keepAccessToSA
+                  )
+                  .map(_ => Redirect(call))
+              case Left(error) =>
+                Future.successful(
+                  handleErrors(error, "[KeepAccessToSAController][continue]")
```

Comment:
if handle errors logs then cool, but if not, dont we want to log here

Reply:
Yes it does log, we pass the string in to add to the logs :-)